middle_end/flambda2/to_cmm/to_cmm.ml

```diff
@@ -454,6 +454,22 @@ let unary_primitive env dbg f arg =
          environment, then [Env.closure_offset] might return [None], even though
          the set of closures has been seen by [To_cmm_closure]. *)
       None, C.unreachable)
+  | Is_boxed_float ->
+    (* As a note, this omits the [Is_in_value_area] check that exists in
+       [caml_make_array], which is used by non-Flambda 2 compilers. *)
```

Comment:
Are you referring to the check in `caml_make_vect` ? The one in `caml_make_array` is slightly different.

I don't know of any good way to reproduce the `Is_in_value_area` check, but if someone has implemented nullable arrays by using a naked zero then this code will segfault when trying to read the tag.

Reply:
If I understand correctly, the `caml_make_array` checks correspond to `Is_flat_float_array`, since its input is supposed to be an array already.

Reply:
Nevermind, I missed the part were it loads the first element and checks its tag... this is indeed the same check as in `caml_make_vect`

Reply:
Not how I'm reading it -- the `Is_in_value_area` test is on the first element of the "template" array (which when the float array optimisation is enabled, is full of boxed floats in the float case).